cdap-client-tests/src/test/java/co/cask/cdap/client/AbstractClientTest.java

```diff
@@ -129,24 +130,57 @@ protected void assertFlowletInstances(ProgramClient programClient, Id.Flow.Flowl
     Assert.assertEquals(numInstances, actualInstances);
   }
 
+  /**
+   * @deprecated since 4.0.0. Please use {@link #assertProgramRunning(ProgramClient, ProgramId)}
+   */
+  @Deprecated
```

Comment:
deprecation in test classes? Are these APIs? Can you just remove this in favor of the other method?